examples/pennylane_run_qaoa_maxcut.py

```diff
@@ -0,0 +1,288 @@
+"""
+.. _qaoa_maxcut:
+
+Quantum Approximate Optimization for MaxCut
+===========================================
+
+In this tutorial, we implement a quantum approximate
+optimization algorithm (QAOA) for the MaxCut problem in Pennylane. (See `Farhi,
+Goldstone, and Gutmann (2014) <https://arxiv.org/abs/1411.4028>`__.) This
+example demonstrates how to sample joint qubit measurements from a variational
+quantum circuit to solve a combinatorial optimization problem.
+"""
+
+##############################################################################
+# 1 Background
+# -------------
+#
+# 1.1 The MaxCut problem
+# ~~~~~~~~~~~~~~~~~~~~~~
+# The aim of MaxCut is to maximize the number of edges (yellow lines) in a graph that are "cut" by
+# a given partition of the vertices (blue circles) into two sets as shown in the figure below.
+#
+# .. figure:: ../../examples/figures/qaoa_maxcut_partition.png
+#    :align: center
+#    :scale: 65%
+#    :alt: qaoa_operators
+#
+#
+# Let's say there are :math:`m` edges. We seek the partition :math:`z` of the vertices into two sets
+# 0 and 1 which maximizes
```

Comment:
I do this with foresight since the computational basis later will come to represent the partitions, e.g. |0101> means that the 0th and 2nd qubits are in set 0, while the 1st and 3rd are in set 1. I perhaps could have labelled the qubits (vertices) as ABCD so there aren't so many 0s and 1s - what do y'all think?

Reply:
The label of the set is not related to the computational basis that represents your partition. So it will be better to keep it very clear. :)

Reply:
Okay I've updated the description of the MaxCut problem and relevant figure to use A and B as the labels for the 2 sets, and elaborated upon the relation between the bitstring z and assignments of each vertex to a set.